libs/ui/design-system/src/lib/switch.tsx

```diff
@@ -1 +1,39 @@
-export { Switch } from '@nmit-coursition/ui/primitives/switch'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+import { Label } from '@nmit-coursition/ui/primitives'
+import * as SwitchPrimitive from '@nmit-coursition/ui/primitives/switch'
+import { cn } from '@nmit-coursition/ui/utils'
+
+/** todo: import util from `typescript` file when mixins architecure is merged  */
+type OverrideProps<T, V> = V & Omit<T, keyof V>
+
+const switchVariants = cva('', {
+  variants: {
+    variant: {
+      /**
+       * OFF: bg-#4A044E, thumb-#FEF3C7
+       * ON: bg-[#FEF3C7], thumb-[#4A044E]
+       */
+      default: '',
+      disabled: '',
```

Comment:
**suggestion:** We shouldn't define any disabled/valid/invalid variants. They should be handled on CSS level.

Reply:
It becomes some difficult to remember the appropriate token for each of the states, but way easier to hook those variants with the component state.

Wdyt?

Reply:
No, if we make it like this, it's possible to enforce it manually, because cva automatically adds it, so it shouldn't be here.
Also it's always prefixed with `disabled:SOMETHING`, so it's easy to see.

And if we use `trim` function, we can split easily into multiple lines and have everything related to one state on one line. :)